be/src/runtime/data_stream_recvr.cc

```diff
@@ -484,7 +487,7 @@ Status DataStreamRecvr::SenderQueue::add_chunks(const PTransmitChunkParams& requ
             }
             _chunk_queue.emplace_back(std::move(item));
         }
-        if (!chunks.empty() && done != nullptr && _recvr->exceeds_limit(total_chunk_bytes)) {
+        if (_chunk_queue.size() > original_size && done != nullptr && _recvr->exceeds_limit(total_chunk_bytes)) {
```

Comment:
Comment for the first condition.

Reply:
Using variable `has_new_chunks` to make it meaningful